### PR TITLE
Propose new recommended community policy R7

### PR DESCRIPTION
**Pull request by ulrikeyang**
Due to a request by the facilities for a well defined space on package information, we are in the process of formulating a new recommended community policy. This is a first preliminary draft to get the process started.

### CODE REVIEW

File: package_policies/R7.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+**R7.** It is recommended that each package should have a README and a LICENSE file in its top directory.
+The README file should contain the following information:
+
+- a link to the package webpage
+
+- contact information, i.e. how to get help (**M5**)
```
Comment (mkstoyanov) [inline package_policies/R7.md]: Do we have to put our e-mails into the public GitHub README?

Isn't it enough that the README has a link to the web-page and the web-page has the contact e-mail.

**Comment by balay** [comment]
Perhaps README.md  name should also acceptable.

Here are some relavent docs wrt README at github, bitbucket, gitlab

https://help.github.com/en/articles/about-readmes
https://confluence.atlassian.com/bitbucket/display-readme-text-on-the-overview-221449772.html
https://docs.gitlab.com/ee/user/markdown.html

Bitbucket is listing a bunch of other acceptable extensions to README. Gitlab mentions a bunch of acceptable locations for this file.

**Comment by maherou** [comment]
I would be inclined to prefer the .md version of these files in order to support formatted text. 

Also, in addition to the README.md and LICENSE.md files, I would like to have us consider files like SUPPORT.md, CONTRIBUTING.md and others.  Facilities staff want to know much more about our products than what they would find in the readme and license files.  Here is some more details about what other people have considered:  https://github.com/kmindi/special-files-in-repository-root/blob/master/README.md

We should come up with a list of these files and then have them reviewed by people like Judy Hill and Katherine Riley and their teams.

**Comment by ulrikeyang** [comment]
I have revised the proposed recommended community policy based on discussions we had at yesterday's xSDK meeting.

**Comment by balos1** [comment]
The short-form of this policy should also be added to the `package_policies/README.md` file (and linked to the R7.md file like the others).

**Review by acfisher — APPROVED** [review]

**Comment by ulrikeyang** [comment]
Rebecca Hartman-Baker of NERSC asks if we could include a pointer to how to install with spack in the installation instructions.

So, we could add '(including pointers on how to install with spack)' after 'information on how to install the package.'

**Comment by maherou** [comment]
It seems to me that we should start with a fairly long list potential informational files.  To address this particular request, could we have a file named INSTALLATION.md or INSTALLING.md or something similar?

Brainstorming on potential files beyond what is already commonly used can be a good part of the discussion.

On Jun 14, 2019, at 12:12 AM, ulrikeyang <notifications@github.com<mailto:notifications@github.com>> wrote:


Rebecca Hartman-Baker of NERSC asks if we could include a pointer to how to install with spack in the installation instructions.

So, we could add '(including pointers on how to install with spack)' after 'information on how to install the package.'

**Comment by ulrikeyang** [comment]
How is installation information handled by other libraries?
Btw, hypre has a INSTALL.md file in the top directory with installation information, so I don’t mind adding INSTALL or INSTALL.md to the list 😊.

In general, I would however prefer to keep the number of files limited.

**Comment by xiaoyeli** [comment]
I am fine with this.  Currently for SuperLU, the installation instructions are also in README.md.

**Review by jthies — APPROVED** [review]

**Comment by balay** [comment]
> Rebecca Hartman-Baker of NERSC asks if we could include a pointer to how to install with spack in the installation instructions.
> 
> So, we could add '(including pointers on how to install with spack)' after 'information on how to install the package.'

I don't think installation instructions is the correct place for spack specific info - the same way most packages don't have ubuntu or fedora, or RHEL - or brew installation instructions.

Once you know these package managers,  all packages are equivalent wrt installation. So its more of learning spack - and 'spack installation instructions' [ spack specific installation instructions within the package are not a substitute].

However - some packages do mention such info on their download web-page. i.e One can get the tarball - and install from source - or use package manager from their favourite distribution to download/install the package etc.

**Comment by maherou** [comment]
Thanks for these comments.  My point is that we should thoroughly explore the space of potential information and then cut down to the essentials.  I think this model of using .md files in the repository is an excellent way to convey basic content about a product.  We should use it for any content that is specific.

Regarding spack, I like @balay comments about homebrew, etc.  Would it be useful to have the Spack package name accessible from one of these files?  For many products, the package name may be obvious, but not for all.

**Review by tzanio — APPROVED** [review]

**Comment by hartwiganzt** [comment]
Maybe it would make sense to note that it is OK  to provide detailed instructions in another file and reference that file from the README.md. But the way it is worded one can argue that this is already permitted.

**Review by mkstoyanov — APPROVED** [review]

File: package_policies/R7.md (hunk 1 above)
Comment (balay) [reply package_policies/R7.md]: I think you can repeat the URL in the 'Contact' section of README

**Comment by balay** [comment]
>  Would it be useful to have the Spack package name accessible from one of these files? For many products, the package name may be obvious, but not for all.

I think its easier to get this [and other spack related] info directly from `spack`

```
$ spack help list
list and search available packages
<snip>
balay@sb /home/balay
$ spack list trilinos
==> 3 packages.
trilinos  trilinos-catalyst-ioss-adapter  xsdktrilinos
balay@sb /home/balay
$ spack list -d petsc
==> 5 packages.
petsc  phist  py-petsc4py  sowing  xsdktrilinos
balay@sb /home/balay
$ spack info py-petsc4py
PythonPackage:   py-petsc4py
<snip>
```
etc..

**Review by xiaoyeli — APPROVED** [review]

**Comment by ulrikeyang** [comment]
I think the full installation information in the README file or a link to where the full installation information is found (which could be a INSTALL file) would be fine. We can reword this to make this clear.

File: package_policies/R7.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+- information on how to install the package or a link to a file (e.g. INSTALL, INSTALL.md or similar) or a website with the installation information
+
+- a link to the package webpage (if there is one)
+
+The LICENSE file should contain the full text of the license. If the full text can be found in a different directory or via a link to a website it is sufficient to provide a link to the full text in the file.
+
+The SUPPORT file should contain the contact information that is required by community policy **M5** to be able to get help.
+
+The CHANGELOG file should contain important changes or a link to a file or webpage with this information, but not each individual commit.
```
Comment (balay) [inline package_policies/R7.md]: `but not each individual commit` -> `but not a list of all individual commit messages`

Comment (balay) [reply package_policies/R7.md]: or:
 `but not each individual commit message`

**Review by balay — APPROVED** [review]

**Review by acfisher — APPROVED** [review]

**Review by mkstoyanov — APPROVED** [review]

**Review by balos1 — CHANGES_REQUESTED** [review]
@ulrikeyang I approve of the changes, but I think the short version of the policy needs to be added to the `package_policies/README.md` file before the PR is merged.

**Comment by balay** [comment]
> @ulrikeyang I approve of the changes, but I think the short version of the policy needs to be added to the `package_policies/README.md` file before the PR is merged.

Perhaps the following change?
```
diff --git a/package_policies/README.md b/package_policies/README.md
index a331169..307b681 100644
--- a/package_policies/README.md
+++ b/package_policies/README.md
@@ -102,6 +102,8 @@ In addition to the required xSDK policies listed above, the following capabiliti
 
 **R6.** [Document versions of packages that it works with or depends upon, preferably in machine-readable form.](/package_policies/R6.md)
 
+**R7.** [Document package information in README and related files.](/package_policies/R7.md)
+
 ## History of the xSDK Community Policies
 
 The original version of this document was prepared in 2015 by Barry Smith with key input from Roscoe Bartlett
```

**Review by balos1 — APPROVED** [review]

**Review by osborn9 — APPROVED** [review]

**Review by jwillenbring — APPROVED** [review]

**Review by luszczek — APPROVED** [review]
